### DISCUSSION TITLE
Adding multiple fragments to the integrated object

### DISCUSSION THREAD

**Discussion by Chhiring-Lama** [stuart-lab/signac#918] [Q&A] [answered]
I added it without validating whether the list of cells are present or not using the following function:

"fragments" is the list of paths to individual fragments, and "patient_sample" is the list of orig.idents. I used the cell list from integrated data as the barcode formatting was different and with barcode from fragments, I was not able to add the fragments to the integrated object.

`
for(i in fragments){
  for (j in patient_sample){
    if (which(fragments == i) == which(patient_sample == j)){
      a <- CreateFragmentObject(
        path = i, 
        cells = cell_list$bc[cell_list$patient == j], 
        validate.fragments = F)
      fragment_paths = append(fragment_paths,a)
    }
  }
}

DefaultAssay(integrated_data) <- "ATAC"
Fragments(integrated_data) <- fragment_paths

Does this work? Should this cause any error downstream for peakcalling or coverageplot? Is there a way to do these without the fragments? Is there a more efficient way to add multiple fragment files to an integrated object?

**Comment by timoast** [comment, marked as answer]
I can't say whether this will cause issues or not without knowing more about the specific objects and how they were created, but you should only set `validate.fragments=FALSE` if you're confident you know what you're doing and what that parameter does.

If you add the fragment file information to each individual object before integrating you shouldn't really need to worry about changing anything after that, and will have access to data from all of the fragment files correctly linked to all the cells in the integrated object.

**Reply by Chhiring-Lama** [reply]
I added fragments to each individual object, but seem to have lost them during integration. So, I am trying to add them back in. I have a dataframe with cell barcode in individual objects and their corresponding names in integrated object which I could use to rename the cells in fragmentobject. Is there a way to do so?

**Reply by timoast** [reply]
The fragments should still be present in the original, uncorrected, assay in the integrated object. For scATAC-seq, I recommend integrating the LSI embeddings rather than the peak matrix itself, as shown here: https://satijalab.org/signac/articles/integrate_atac.html (in which case there is no integrated assay, just an integrated dimension reduction).

**Reply by Chhiring-Lama** [reply]
Thank you for the quick response and providing necessary resources!

They are multiomic samples which I had created seurat object for individually, but someone else integrated them. The final integrated object does not contain the fragments anymore. I want to add the fragments on my own, if I can, for this time as I need to produce an output for an analysis and can talk to them about integrating LSI embeddings for next run onwards. I was wondering if I can edit the fragment file themselves in R to match the format from integrated object. If not, reintegrating them is always an option too. Thanks!

**Reply by timoast** [reply]
You can add the fragment objects to the integrated object but you need to get the mapping of cell names in the object to names in the fragment files correct. If there were any duplicated cell names across the object that were integrated, the cells will have been renamed and will no longer match those in the fragment file. If you can work out how to correctly map the cell names, it should work fine. We handle this automatically for you if you merge multiple chromatin assays, which is why it might be easier just to create a merged assay.
